lib/ansible/plugins/inventory/linode.py

```diff
@@ -125,18 +131,31 @@ def _filter_by_config(self, regions, types):
                 if instance.type.id in types
             ]
 
-    def _add_instances_to_groups(self):
+    def _add_instances_to_groups(self, use_public_ip):
         """Add instance names to their dynamic inventory groups."""
         for instance in self.instances:
-            self.inventory.add_host(instance.label, group=instance.group)
+            if use_public_ip:
+                dest = self._get_instance_public_ip(instance)
+            else:
+                dest = instance.label
+            self.inventory.add_host(dest, group=instance.group)
+
+    def _get_instance_public_ip(self, instance):
+        """Returns a the public IP address of the instance"""
+        return [ip for ip in instance.ipv4 if not ipaddress.ip_address(ip).is_private][0]
 
-    def _add_hostvars_for_instances(self):
+    def _add_hostvars_for_instances(self, use_public_ip):
         """Add hostvars for instances in the dynamic inventory."""
         for instance in self.instances:
+            if use_public_ip:
+                dest = self._get_instance_public_ip(instance)
```

Comment:
Again, can we name this `dest` better? Will help readability as the module grows.